eclair-core/src/main/scala/fr/acinq/eclair/channel/TxPublisher.scala

```diff
@@ -0,0 +1,418 @@
+/*
+ * Copyright 2021 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.channel
+
+import akka.actor.typed.eventstream.EventStream
+import akka.actor.typed.scaladsl.adapter.TypedActorRefOps
+import akka.actor.typed.scaladsl.{ActorContext, Behaviors}
+import akka.actor.typed.{ActorRef, Behavior}
+import fr.acinq.bitcoin.{ByteVector32, ByteVector64, Crypto, Satoshi, Script, Transaction, TxOut}
+import fr.acinq.eclair.NodeParams
+import fr.acinq.eclair.blockchain.bitcoind.rpc.ExtendedBitcoinClient
+import fr.acinq.eclair.blockchain.bitcoind.rpc.ExtendedBitcoinClient.FundTransactionOptions
+import fr.acinq.eclair.blockchain.fee.FeeratePerKw
+import fr.acinq.eclair.blockchain.{CurrentBlockCount, WatchConfirmed, WatchEvent, WatchEventConfirmed}
+import fr.acinq.eclair.transactions.Transactions._
+import fr.acinq.eclair.transactions.{Scripts, Transactions}
+import fr.acinq.eclair.wire.protocol.UpdateFulfillHtlc
+
+import java.util.concurrent.Executors
+import scala.collection.immutable.SortedMap
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+ * Created by t-bast on 25/03/2021.
+ */
+
+/**
+ * This actor ensures its parent channel's on-chain transactions confirm in a timely manner.
+ * It sets the fees, tracks confirmation progress and bumps fees if necessary.
+ */
+object TxPublisher {
+
+  // @formatter:off
+  sealed trait Command
+  sealed trait PublishTx extends Command {
+    /** Actor that should receive a [[WatchEventConfirmed]] once the transaction has been confirmed. */
+    def replyTo: ActorRef[WatchEvent]
```

Comment:
This field looks unused, maybe let's defer introducing it before it's really used? Also why doesn't it take a `WatchEventConfirmed`?

```suggestion
    def replyTo: ActorRef[WatchEventConfirmed]
```

Reply:
I figured it's a good practice to define it right from the start, but maybe it's worth waiting to actually use it indeed.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1749/commits/d17b3d0ab1d33f504c04ad67a537b46649ee3230